wcfsetup/install/files/lib/util/HeaderUtil.class.php

```diff
@@ -106,6 +107,28 @@ public static function sendNoCacheHeaders()
         @\header('Cache-Control: max-age=0, no-cache, no-store, must-revalidate');
     }
 
+    /**
+     * Returns an response matching the given response with headers preventing in-browser
+     * caching attached.
+     */
+    public static function withNoCacheHeaders(ResponseInterface $response): ResponseInterface
```

Comment:
I wonder if this should be moved somewhere to `lib/system/io/http/`?

Reply:
This would only make sense if you're looking forward to phase out this class?

Reply:
Looking at the methods in `HeaderUtil`:

- `setCookie`: This is one of the issues that should be solved with PSR-7. Will need to be adapted, might be replaced by: https://github.com/dflydev/dflydev-fig-cookies
- `getCookieDomain`: Might be simplified once multi-domain setups are gone?
- `sendHeaders`: Will become obsolete with PSR-7 and is then better solved using a middleware.
- `sendNoCacheHeaders`: This is replaced by the method we are currently discussing.
- `parseOutput`: Is intimately tied to `sendHeaders`.
- `redirect`: Will be replaced by `RedirectResponse`.
- `delayedRedirect`: Should probably be replaced by a specific `DelayedRedirectResponse`.

Basically all of the methods in `HeaderUtil` will become obsolete with the migration to PSR-7. Thus why I want to discuss whether re-architecting this might make sense. For Guzzle we already have the RedirectGuard as a single-purpose class:

https://github.com/WoltLab/WCF/blob/master/wcfsetup/install/files/lib/system/io/http/RedirectGuard.class.php

Reply:
I agree, spreading things out while still having a mix of everything certainly does not improve the situation.